### DISCUSSION TITLE
http post application/octet-stream

### DISCUSSION THREAD

**Discussion by JamieMcQuire23** [JAndrassy/WiFiEspAT#61] [Q&A]
Hi, with the client.write() function can we send a .bin file directly from a SD card to a web serve. I have been trying but coming across a 422 error.

**Comment by JAndrassy** [comment]
do you use client.write(file)?

**Reply by JamieMcQuire23** [reply]
Yeah. I send a SdFat.h SdFile binary data file.

**Reply by JamieMcQuire23** [reply]
@jandrassy with the client.write(file) function will the data be wrote as bytes?

**Reply by JAndrassy** [reply]
yes

**Reply by JAndrassy** [reply]
did you try the SDWebServer example? Put a jpg picture on the SD card and try to show it in the browser

**Reply by JamieMcQuire23** [reply]
@jandrassy found the error was due to 2 reasons... First, my header was incorrect (needed organised into form data). Second, It appears that (SdFat.h) SdFile is incompatible with this library. Do you have a work around at hand? Otherwise I will start looking at how I can get the client.write function to work with SdFat.

Edit: **Solved** easy solution for anyone who finds this... <SDFat.h> has backwards compatibility with the <SD.h> file type. Sorted with a large data transfer now - speed isn't where I would like it (6kbps) but will experiment with adjusting the baud.
